### ISSUE TITLE
v0.4.4 slow and unresponsive for 1GB TIF

### ISSUE THREAD

**Issue by Mrc010** [RENCI/Segmentor#138]
Hi,

thanks for your interesting tool. I am having problems of the following kind:
When I load a 1GB image stack (TIF) and its corresponding masks (also 1GB TIF), the process is quite memory hungry (speaking of > 100GB) and loading takes hours. Also, when the loading process is eventually over, any interaction with the GUI make it unresponsive.

**Comment by davidborland** [comment]
Yes, the tool has typically been used for much smaller volumes. Most of our data are 128x128x64 volumes (~2MB), with ~1000 cells per volume.

Unfortunately, I'm not surprised that there are performance issue for much larger volumes.

**Comment by kevinjohncutler** [comment]
I have the same issue. There are other programs that successfully load these image sizes (e.g. vispy/napari) without these memory issues, so optimization is certainly possible (but might be too much work to implement here).
